src/Controller/JobController.php

```diff
@@ -0,0 +1,338 @@
+            $this->jobValidator->nameIsValid($name);
+            $rows = $this->jobsRepository->selectByNameLike($name);
+            $jobsArr = [];
+            foreach ($rows as $job) {
+                $jobsArr[] = [
+                    'id' => $job->getId(),
+                    'name' => $job->getName(),
+                    'description' => $job->getDescription(),
+                    'active' => $job->getActive(),
+                    'priortiy' => $job->getPriority(),
+                    'createdAt' => $job->getCreatedAt(),
+                    'company' => [
+                        'id' => $job->getCompany()->getId(),
+                        'company' => $job->getCompany()->getName()
+                        
+                    ]
+                ];
+            }
+            
+            return new JsonResponse([
+                'results' => [
+                    'error' =>false,
+                    'jobs' => $jobsArr
+                ]
+            ]);
+        } catch (\Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * updates a job from the database
+     *
+     * @param  mixed $id
+     * @param  mixed $request
+     * @return Response
+     */
+    public function updateJob(int $id, Request $request): Response
+    {
+        try {
+            $this->jobValidator->idIsValid($id);
+            $name = $request->get('name');
+            if (!empty($name)) {
+                $this->jobValidator->nameIsValid($name);
+            }
+            
+            $requestParams = $request->query->all();
+            $updateResult = $this->jobsRepository->update($id, $requestParams);
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => false,
+                        'rows_updated' => $updateResult
+                    ]
+                ]
+            );
+        } catch (\Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * delete a job from the database
+     *
+     * @param  mixed $id
+     * @return Response
+     */
+    public function deleteJob(int $id): Response
+    {
+        try {
+            $this->jobValidator->idIsValid($id);
+            $jobToDelete = $this->jobsRepository->find($id);
+            
+            if (is_null($jobToDelete)) {
+                throw new \InvalidArgumentException('Could not find job with id: ' . $id);
+            }
+
+            $this->jobsRepository->remove($jobToDelete);
+            return new JsonResponse([
```

Comment:
This is the response returned if delete successful:
![image](https://user-images.githubusercontent.com/19302124/207557293-bc7bf41e-42af-4a61-85d9-8bb155ca2151.png)
 
$deletedId is not set and empty($deletedId) will be true and !empty($deletedId) will be false. Put in the $deletedId variable the id of the job to delete before the deletion at line 320. $deletedId =  $jobToDelete->getId() and replace in the response  'deleted' => !empty($deletedId), with job  'job_deleted_id' => $deletedId